internal/engine/image_build_and_deployer.go

```diff
@@ -215,6 +214,24 @@ func (ibd *ImageBuildAndDeployer) deploy(ctx context.Context, st store.RStore, p
 	return ibd.k8sClient.Upsert(ctx, newK8sEntities)
 }
 
+// We can also skip the push of the image if it isn't used
+// in any k8s resources! (e.g., it's consumed by another image).
+func (ibd *ImageBuildAndDeployer) canSkipPushOfTarget(iTarget model.ImageTarget, kTargets []model.K8sTarget) bool {
```

Comment:
nit: this comment belongs above the codeblock at ln 224, not above the whole function